libr/anal/fcn.c

```diff
@@ -738,6 +739,14 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 len, int
 					last_is_mov_lr_pc = true;
 				}
 			}
+			if (has_stack_regs) {
+				bool has_dst_reg = op.dst && op.dst->reg && op.dst->reg->name;
```

Comment:
Could you please extract those into the separate `inline` functions? To prevent the code duplication.